compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPULowerExecutableTarget.cpp

```diff
@@ -199,7 +199,7 @@ void LLVMCPULowerExecutableTargetPass::runOnOperation() {
   }
   case IREE::Codegen::DispatchLoweringPassPipeline::CPUDataTiling: {
     TilingConfig tilingConfig = getTilingConfigForPipeline(moduleOp);
-    addCPUDataTilingPipeline(pipeline, tilingConfig, enableVectorMasking);
+    addCPUDataTilingPipeline(pipeline, tilingConfig);
```

Comment:
Can we try to fix the conditions about enableVectorMasking? You will hit the same issue in fusion. In that case, you will enforce masking in MultiTilingExpertPassPipeline, which seems a big hammer to me. 

```
  bool enableVectorMasking = isX86(target) || isRISCV(target) ||
                             (isAArch64(target) && hasAnySVEFeature(target));
```

Reply:
Not sure I follow. In other pipelines, and for Linalg ops, we can apply peeling so we wouldn't need masking... 

I can keep the flag and just pass true from the caller, if that looks better

Reply:
I thought that we are moving from peeling to masking? Also, see the below pipeline. We don't use peeling for multi level tiling expert.

https://github.com/openxla/iree/blob/cddd084794307ef79a30096a4c3a921a5e041cea/compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPULowerExecutableTarget.cpp#L171-L177

Reply:
Well, it depends on the operation. For non-DT matmuls we use peeling by default and I would expect something similar for other ops for targets that don't support masking but I haven't hit any issue during profiling that motivates that. IOW, we may want to replace that `false` with `!enableVectorMasking`, but that would need some testing.

Reply:
It's not only used by nonDT matmul. It is also used by reduction and element wise ops. You will see the same issue in this pipeline if the dispatch has elementwise+pack.